grid_map_visualization/src/GridMapVisualization.cpp

```diff
@@ -6,165 +6,170 @@
  *	 Institute: ETH Zurich, ANYbotics
  */
 
-#include "grid_map_visualization/GridMapVisualization.hpp"
 #include <grid_map_core/GridMap.hpp>
 #include <grid_map_ros/GridMapRosConverter.hpp>
 
-using namespace std;
-using namespace ros;
+#include <chrono>
+#include <memory>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "grid_map_visualization/GridMapVisualization.hpp"
 
-namespace grid_map_visualization {
+namespace grid_map_visualization
+{
 
-GridMapVisualization::GridMapVisualization(ros::NodeHandle& nodeHandle,
-                                           const std::string& parameterName)
-    : nodeHandle_(nodeHandle),
-      visualizationsParameter_(parameterName),
-      factory_(nodeHandle_),
-      isSubscribed_(false)
+GridMapVisualization::GridMapVisualization(const std::string & parameterName)
+: visualizationsParameter_(parameterName)
+  // isSubscribed_(false)
 {
-  ROS_INFO("Grid map visualization node started.");
+  node_ = std::make_shared<rclcpp::Node>("grid_map_visualization");
```

Comment:
Why not make the grid map visualization class derive from node? That's pretty typical structure

Reply:
The visualisation classes (i.e [FlatPointCloudVisualization](https://github.com/Marwan99/grid_map/blob/visualization-port/grid_map_visualization/src/visualizations/FlatPointCloudVisualization.cpp) ..etc) needed access to the node pointer so they can declare/read parameters, log, check number of subscribers and create publishers. 

I was initially going down the route of making the grid map visualization class derive from node but I was unable to get a pointer to the node created to pass it on to the [visualisations classes](https://github.com/Marwan99/grid_map/tree/visualization-port/grid_map_visualization/src/visualizations), and the only solution I could find was to pass
 `rclcpp::node_interfaces::NodeLoggingInterface`, `rclcpp::node_interfaces::NodeGraphInterface`, `rclcpp::node_interfaces::NodeParametersInterface` and `rclcpp::node_interfaces::NodeBaseInterface` to each [visualization class](https://github.com/Marwan99/grid_map/tree/visualization-port/grid_map_visualization/src/visualizations) to be able to perform the things mentioned above. So instead I went with creating the node object inside the class to be be able to pass it around.

What do you think?

Reply:
https://github.com/Marwan99/grid_map/blob/visualization-port/grid_map_visualization/include/grid_map_visualization/visualizations/MapRegionVisualization.hpp#L29 they derive from the base class, if that's a node, then each of the visualization is then itself a node.